sqlglot/optimizer/pushdown_predicates.py

```diff
@@ -18,58 +20,116 @@ def pushdown_predicates(expression):
     Returns:
         sqlglot.Expression: optimized expression
     """
+    expression = simplify(expression)
+
     for scope in reversed(traverse_scope(expression)):
         where = scope.expression.args.get("where")
 
         if not where:
             continue
 
         condition = where.this.unnest()
-        predicates = (
-            condition.flatten() if isinstance(condition, exp.And) else [condition]
-        )
 
-        for predicate in predicates:
-            sources = [
-                scope.sources.get(table) for table in exp.column_table_names(predicate)
-            ]
+        cnf_like = normalized(condition) or not normalized(condition, dnf=True)
+
+        predicates = list(
+            condition.flatten()
+            if isinstance(condition, exp.And if cnf_like else exp.Or)
+            else [condition]
+        )
 
-            if len(sources) != 1:
-                continue
+        if cnf_like:
+            for predicate in predicates:
+                for node in nodes_for_predicate(predicate, scope).values():
+                    if isinstance(node, exp.Join):
+                        predicate.replace(exp.TRUE)
+                        node.on(predicate, copy=False)
+                        break
```

Comment:
Why break?

Reply:
we only want to push it down to one join

Reply:
I know some folks are anti-comments...
But I think little comments like this are helpful, especially with gnarly code like this.